pipe_anchorages/transforms/smart_thin_records.py

```diff
@@ -68,7 +81,7 @@ def thin(self, grouped_records, anchorage_map):
                 and rcd.timestamp - last_rcd.timestamp >= self.min_gap
             ):
                 active.add(last_rcd)
-                active.add(rcd)
+                active.add(rcd._replace(is_possible_gap_end=True))
 
             if self.transition_map[(last_state, state)]:
```

Comment:
My first thought is that at this point the record is still being dropped due to thinning (if both points - before and after gap - are `IN_PORT`) , no?

Reply:
@Chr96er , if there is a gap seen during the thinning it should keep the start and end of the gap in the thinned points, and (this is the key part) the end of the gap gets tagged with `is_possible_gap_end`. Then in `creare_in_out_events`, points with that tag, but no other points can emit gaps. So we should still get gaps when there's a legitimate gap.

It might make sense to only emit this pair of points when the current point is in port (since that's the only time we'd generate a gap), but I'm not sure it makes a difference one way or the other.